index.js

```diff
@@ -112,6 +122,52 @@ class Conf {
 		}
 	}
 
+	_migrate(migrations, versionToMigrate = '0.0.0') {
+		const MIGRATION_KEY = '__conf-migrated-version__';
+
+		const previousMigratedVersion = this.get(MIGRATION_KEY, '0.0.0');
+
+		const newerVersions = Object.keys(migrations)
+			.filter(version => this._shouldPerformMigration(version, previousMigratedVersion, versionToMigrate))
+			.sort(semver.compare);
+
+		newerVersions.forEach(version => {
+			try {
+				const migration = migrations[version];
+				migration(this);
+
+				this.set(MIGRATION_KEY, version);
+			} catch (error) {
+				console.error('Something went wrong during the migration!');
```

Comment:
What you guys think about providing information with console.error?

If you agree with this, should we add a stub in the tests to prevent this from being printed?

![image](https://user-images.githubusercontent.com/24696635/61673171-53ce5f00-ace6-11e9-8e9c-2662c1fa86c0.png)

Reply:
I'm fine with the `console.error` or `console.warn`, but I will leave that to @sindresorhus

Reply:
Actually I've gave a second though about this and perhaps logging is not a good idea. This library is being used by a lot of CLI tools and having control over logging is crucial. Having a library perform logging is not ideal, specially if we don't have any control over it.

Anyway, lets see what @sindresorhus has to say.

Reply:
Do you have any recommendation/preference for the stubbing library?

Reply:
I usually use Jest's built-in way of stubbing/mocking. But since we are not using Jest, we may use sinon or something. But I would wait for @sindresorhus feedback as it might not be necessary at all if we remove the logging.

Reply:
I don't think we should catch migration errors. If migrations fail, the conf state will not be in the expected state and other things will probably randomly fail too.

Reply:
I guess he's just catching to have a more friendly message and throwing the error again.

Reply:
What @satazor said 😄 

If you're fine with the current solution, can I choose a stubbing library or do you prefer one?

If you prefer to just throw the error without a friendlier message, please disregard my last question.

Reply:
I missed that you were throwing the error too. I would prefer the best of both worlds and modify the original error with a more useful message, but also include the original message in the error. So no `console.error` logging.

Reply:
Would be useful to also include which step in the migration that failed.

Reply:
As for stubbing, https://sinonjs.org/ is good.

Reply:
Your suggestion makes total sense. Updated it 😄